src/TestRunner.cpp

```diff
@@ -49,68 +31,64 @@ bool TestRunner::runTests(MappedMethods expectedResult,
         errs() << " **FAILED**\n";
         testPassed = EXIT_FAILURE;
       }
-      errs() << "EXPECTED {" << pass << "}\n";
-      errs() << "RECEIVED {" << analysisPass << "}\n\n";
+      errs() << "EXPECTED " << pass << "\n";
+      errs() << "RECEIVED " << analysisPass << "\n\n";
     }
   }
 
   return testPassed;
 }
 
-MappedMethods TestRunner::buildExpectedResults(std::string testName,
-                                               std::string passName) {
+FunctionMappedMethods TestRunner::buildExpectedResults(std::string testName,
+                                                       std::string passName) {
   std::ifstream testFile("../test/" + testName + ".txt");
   std::string line;
-  MappedMethods expectedResult;
+  FunctionMappedMethods expectedResult;
 
   if (testFile.is_open()) {
     while (std::getline(testFile, line)) {
-      if (line.size() > 14 && line.substr(0, 14) == "ALLOW_REDEFINE") {
-        continue;
-      }
-
-      int spaceCount = 0;
-      std::string branchName;
-      std::string varName;
-      std::set<std::string> calledMethodsSet;
-      std::string currentCM;
-      std::string passType;
-
-      for (char c : line) {
-        if (c == ' ') {
-          spaceCount++;
-          continue;
-        }
-
-        if (spaceCount == 0) {
-          passType += c;
-        } else if (spaceCount == 1) {
-          if (passType != passName) {
-            break;
-          }
-
-          branchName += c;
-        } else if (spaceCount == 2) {
-          varName += c;
-        } else if (spaceCount == 3) {
-          if (c == '{') {
-            continue;
-          }
-          if (c == ',' || c == '}') {
-            if (currentCM.size() > 0)
-              calledMethodsSet.insert(currentCM);
-            currentCM = "";
-            continue;
-          }
-          currentCM += c;
+      // we will be assuming our test case writing is correctly formatted.
+      // test arguments are written in one line and can be put in any order.
+      // they look like the following:
+      // pass=<str> function=<str> branch=<str> var=<str> methods=<set<str>>
+      // e.g., pass=MustCall function=main branch=entry var=s methods={free}.
+      // there must not be any spaces anywhere else, including in methods
+      // argument. to add more methods to method arg, it would look like:
+      // methods={x,y}.
```

Comment:
If one of these comments is incorrectly formatted, we should at least try to fail politely - while writing test cases, we're all likely to make a mistake in formatting one of these occasionally, so it'd be best if that doesn't trigger a segfault.